src/mono/wasm/host/wasi/WasiEngineHost.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Maybe wasi host should be separate project and separate host. 
There are dependencies which don't make sense for WASI like `BrowserDebugHost`

Reply:
We can discuss that. For now I want to have a single WasmAppHost which can run any of the projects.